rflx/pyrflx/typevalue.py

```diff
@@ -226,39 +239,611 @@ class OpaqueValue(TypeValue):
+            return self.__simplified(Add(self._fields[prv].first, self._fields[prv].length))
+        return UNDEFINED
+
+    def _has_first(self, fld: str) -> bool:
+        return isinstance(self._get_first_unchecked(fld), Number)
+
+    def _get_first(self, fld: str) -> Number:
+        first = self._get_first_unchecked(fld)
+        assert isinstance(first, Number)
+        return first
+
+    @property
+    def name(self) -> str:
+        return self._model.name
 
     @property
     def accepted_type(self) -> type:
         return bytes
 
+    @property
+    def value(self) -> Any:
+        raise NotImplementedError
+
     @property
     def literals(self) -> Mapping[Name, Expr]:
         return {}
+
+    def assign(self, value: bytes, length: int = 0) -> None:
+        """
+        :param value: Bitstring representation of the message
+        :param length: use length parameter if complete
+        length of the message is less than 1 Byte
+        :return:
+        """
+
+        msg_as_bitstr: Bitstring = Bitstring().from_bytes(value)
+        self.assign_bitvalue(msg_as_bitstr, length)
+
+    def assign_bitvalue(self, value: Bitstring, length: int = 0) -> None:
+        """
+        :param value: Bitstring representation of the message
+        :param length: use length parameter if complete
+        length of the message is less than 1 Byte
+        """
+
+        current_field_name = self._next_field(model.INITIAL.name)
+        field_first_in_bitstr = 0
+        field_length = 0
+
+        while current_field_name != model.FINAL.name and (
+            field_first_in_bitstr + field_length
+        ) <= len(str(value)):
+
+            current_field = self._fields[current_field_name]
+            if isinstance(current_field.typeval, OpaqueValue) and not self._has_length(
+                current_field_name
+            ):
+
+                self._fields[current_field_name].first = self._get_first(current_field_name)
+                self.set(current_field_name, value[field_first_in_bitstr:])
+                self._fields[current_field_name].length = Number(current_field.typeval.length)
+            else:
+                assert isinstance(current_field.length, Number)
+                field_length = current_field.length.value
+
+                if field_length < 8:
+
+                    if length != 0 and length < 8:
+                        field_first_in_bitstr = field_first_in_bitstr + 8 - field_length
+
+                    self.set(
+                        current_field_name,
+                        value[field_first_in_bitstr : field_first_in_bitstr + field_length],
+                    )
+                    field_first_in_bitstr = field_first_in_bitstr + field_length
+
+                elif field_length >= 9 and field_length % 8 != 0:
+                    number_of_bytes = field_length // 8 + 1
+                    field_bits = Bitstring()
+
+                    for _ in itertools.repeat(None, number_of_bytes - 1):
```

Comment:
What is the rationale to use `itertools.repeat` here? I'm not sure if it is the right choice for what you want to do.

Reply:
Fixed in https://github.com/Componolit/RecordFlux/pull/198/commits/1ab7b6311e8bc2f82f25c3c4b96d70a135337223.